### PR TITLE
reaxff/species delete rate limit

### PR DESCRIPTION
**Pull request by jrgissing**
**Summary**

add 'delete_rate_limit' keyword to set upper limit on rate of molecule deletion

**Related Issue(s)**

none

**Author(s)**

Jake Gissinger (NASA)

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

**Implementation Notes**

unlike the related 'fix bond/react' keyword, smooth restart support has not been implemented

**Post Submission Checklist**

- [ ] The feature or features in this pull request is complete
- [ ] Licensing information is complete
- [ ] Corresponding author information is complete
- [ ] The source code follows the LAMMPS formatting guidelines
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [ ] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

**Further Information, Files, and Links**

### CODE REVIEW

File: src/REAXFF/fix_reaxff_species.cpp [hunk 1]
```diff
@@ -50,6 +64,8 @@ FixReaxFFSpecies::FixReaxFFSpecies(LAMMPS *lmp, int narg, char **arg) :
     x0(nullptr), BOCut(nullptr), fp(nullptr), pos(nullptr), fdel(nullptr), ele(nullptr),
     eletype(nullptr), filepos(nullptr), filedel(nullptr)
 {
+  if (lmp->citeme) lmp->citeme->add(cite_reaxff_species_delete);
```
Comment (akohlmey) [inline src/REAXFF/fix_reaxff_species.cpp]: Please see: https://docs.lammps.org/stable/Modify_style.html#citation-reminder-suggested

**Review by akohlmey — CHANGES_REQUESTED** [review]
I didn't notice before, but since we have a KOKKOS version of `fix reaxff/species`, this feature and support for the "delete" keyword, which it depends on and that was added previously, need to be ported to the KOKKOS version.

**Comment by jrgissing** [comment]
> I didn't notice before, but since we have a KOKKOS version of `fix reaxff/species`, this feature and support for the "delete" keyword, which it depends on and that was added previously, need to be ported to the KOKKOS version.

last time, we decided that the 'delete' keyword does not affect KOKKOS, because only one major routine (FindMolecule) is overridden by KOKKOS, and I do not modify that one.

**Comment by akohlmey** [comment]
> last time, we decided that the 'delete' keyword does not affect KOKKOS, because only one major routine (FindMolecule) is overridden by KOKKOS, and I do not modify that one.

I apologize. This is my bad. Didn't notice and didn't recall.

Too many pull requests and too much hacking on different things for an old and abused brain like mine, I suppose. 

Jumping between rather different tasks like CMake scripting, grammar and overview checks of the manual, portability checks (and bootstrapping all kinds of virtual machines and cross-compilers for that), and testing GPU kernels on different OpenCL hardware seems to be taking its toll. :-(

**Review by akohlmey — APPROVED** [review]

**Review by sjplimp — APPROVED** [review]

**Pull request merged**